naff/client/client.py

```diff
@@ -1752,90 +1752,128 @@ async def _dispatch_interaction(self, event: RawGatewayEvent) -> None:
+    @Listener.create("raw_message_create", is_default_listener=True)
+    async def _dispatch_prefixed_commands(self, event: RawGatewayEvent) -> None:
         """Determine if a prefixed command is being triggered, and dispatch it."""
-        message = event.message
+        # don't waste time processing this if there are no prefixed commands
+        if not self.prefixed_commands:
+            return
+
+        data = event.data
 
-        if not message.content:
+        # many bots will not have the message content intent, and so will not have content
+        # for most messages. since there's nothing for prefixed commands to work off of,
+        # we might as well not waste time
+        if not data.get("content"):
             return
 
-        if not message.author.bot:
-            prefixes: str | Iterable[str] = await self.generate_prefixes(self, message)
+        # webhooks and users labeled with the bot property are bots, and should be ignored
+        if data.get("webhook_id") or data["author"].get("bot", False):
+            return
 
-            if isinstance(prefixes, str) or prefixes == MENTION_PREFIX:
-                # its easier to treat everything as if it may be an iterable
-                # rather than building a special case for this
-                prefixes = (prefixes,)  # type: ignore
+        # now, we've done the basic filtering out, but everything from here on out relies
+        # on a proper message object, so now we either hope its already in the cache or wait
+        # on the processor
 
-            prefix_used = None
+        # first, let's check the cache...
+        message = self.cache.get_message(int(data["channel_id"]), int(data["id"]))
 
-            for prefix in prefixes:
-                if prefix == MENTION_PREFIX:
-                    if mention := self._mention_reg.search(message.content):  # type: ignore
-                        prefix = mention.group()
-                    else:
-                        continue
+        # this huge if statement basically checks if the message hasn't been fully processed by
+        # the processor yet, which would mean that these fields aren't fully filled
+        if message and (
+            (not message._guild_id and event.data.get("guild_id"))
+            or (message._guild_id and not message.guild)
+            or not message.channel
+        ):
+            message = None
 
-                if message.content.startswith(prefix):
-                    prefix_used = prefix
-                    break
+        # if we didn't get a message, then we know we should wait for the message create event
+        if not message:
+            try:
+                # i think 10 seconds is a very generous timeout limit
+                event: MessageCreate = await self.wait_for(
+                    MessageCreate, checks=lambda e: int(e.message.id) == int(data["id"]), timeout=10
+                )
```

Comment:
10 seconds is a bit overkill. 2 seconds should always be the response target for an application that wants to feel responsive. Better to fail outright than to sit and wait for 10 seconds 
```suggestion
                # i think 2 seconds is a very generous timeout limit
                event: MessageCreate = await self.wait_for(
                    MessageCreate, checks=lambda e: int(e.message.id) == int(data["id"]), timeout=2
                )
```

Reply:
Fair, I just picked a random number anyways.